src/de/espend/idea/laravel/stub/processor/BladeCustomDirectivesVisitor.java

```diff
@@ -0,0 +1,69 @@
+package de.espend.idea.laravel.stub.processor;
+
+import com.intellij.openapi.util.Pair;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiRecursiveElementVisitor;
+import com.jetbrains.php.lang.psi.elements.Method;
+import com.jetbrains.php.lang.psi.elements.MethodReference;
+import com.jetbrains.php.lang.psi.elements.StringLiteralExpression;
+import fr.adrienbrault.idea.symfony2plugin.Symfony2InterfacesUtil;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Consumer;
+
+public class BladeCustomDirectivesVisitor extends PsiRecursiveElementVisitor {
+
+    private Set<String> availableClasses = new HashSet<>();
+
+    private Consumer<Pair<PsiElement, String>> consumer;
+
+    public BladeCustomDirectivesVisitor(@NotNull Consumer<Pair<PsiElement, String>> consumer) {
+        this.consumer = consumer;
+
+        availableClasses.add("\\Illuminate\\Support\\Facades\\Blade");
+        availableClasses.add("\\Blade");
+        availableClasses.add("Blade");
+    }
+
+    @Override
+    public void visitElement(PsiElement element) {
+
+        try {
+            if (!(element instanceof MethodReference)) {
+                return;
+            }
+
+            MethodReference methodReference = (MethodReference) element;
+            PsiElement[] parameters = methodReference.getParameters();
+
+            if (parameters.length == 0) {
+                return;
+            }
+
+            if (!"directive".equals(methodReference.getName())) {
+                return;
+            }
+
+            Method method = Symfony2InterfacesUtil.getMultiResolvedMethod(methodReference);
```

Comment:
you are not allowed to use `resolve` in index process. the index process make it possible to resolve such methods so as you can see there is a conflict in intellij api usage here. :)

i think this works in you case because `getMultiResolvedMethod` does not leave the current file document. if you extend more classes in PhpClass itself you will run it this issue.

you are just allowed to use document. so try to match against some method name or class usage. for example https://github.com/Haehnchen/idea-php-symfony2-plugin/blob/master/src/fr/adrienbrault/idea/symfony2plugin/stubs/indexes/AnnotationRoutesStubIndex.java. it collect all `@Route` php annotations